h2o-algos/src/main/java/hex/glm/GLM.java

```diff
@@ -3856,6 +3854,26 @@ public void setNonNegative() {
       }
     }
 
+    public void setNonNegative(Frame otherConst) {
+      List<String> constraintNames = extractVec2List(otherConst);
```

Comment:
Can we afford to fit contraintsNames in the List? What is the upper bound to numRows? When I see something depending on numRows exact from Vec it raises a red flag for me :)

Reply:
@valenad1 The constraints will depend on the number of predictors in a model.  It does not depends on dataset length.  In addition, we only do it once if beta_constraint is enabled.

Reply:
you mean size of X right? Then it is ok, I would just comment why it is ok in the code